google/cloud/security/inventory/pipeline_builder.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Builds the inventory pipelines to run, and in the correct order to run."""
+
+import anytree
+import importlib
+import sys
+
+from google.cloud.security.common.util import file_loader
+from google.cloud.security.common.util import log_util
+
+
+LOGGER = log_util.get_logger(__name__)
+
+
+class PipelineBuilder(object):
+    """Inventory Pipeline Builder."""
+
```

Comment:
To my earlier comment regarding how a user knows about what resources are available. We could create a flag for inventory that (--list_resources) which prints the a list of resources that you store here? That would be super helpful.